weareinreach/inreach-catalog
# Feature/1197433510679486/mexico map

## Description

Adds a custom map overlay to the map on the mexico home page
The map uses generalised crime data to color Mexican states based on the value provided for that state

Components added:
- AsylumConnectMexicoMapContainer.js
    - contains the crime map, legend and map type indicator
- AsylumConnectMexicoCrimeMap.js
    - displays the shapes for the map overlay. Maps over data to create custom AC Map Polygons with info windows
- AsylumConnectMapPolygon.js
    - Custom implementation of Google Maps Polygon

 Files Added:
- src/data/mexicoCrimeData.js
    - contains:
          - name of location
          - generalised crime data
          - coordinates for location centre
          - coordinates to draw polygon shape

- Asana ticket: https://app.asana.com/0/1132189118126148/1197433510679486/f

## PR Checklist




## How to Test

1. Navigate to AC catalog home page
2. Select 'Mexico' from the country dropdown
3. Verify that the map shows by default on the search page
4. Click on a Mexican state 
5. Verify that info window is shown with the name of the Mexican state and the generalised crime rate for the state
6. Click on the state again
7. Verify that info window is closed
8. Click on a Mexican state
9. Click the 'x' on the info window
10. Verify that info window is closed
11. Type a location into the search bar
12. Click 'Search'
13. Verify that map is not shown if search returns results

1. Navigate to AC catalog home page
2. Select United States or Canada from the country dropdown
3. Verify that the map is not shown on the search page

![mexico-map](https://user-images.githubusercontent.com/7406914/97368531-8a980b00-1881-11eb-8eeb-2f73493cc5a3.jpeg)
